services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py

```diff
@@ -0,0 +1,119 @@
+"""this module creates a background task that monitors changes in the database.
+First a procedure is registered in postgres that gets triggered whenever the outputs
+of a record in comp_task table is changed.
+"""
+
+import asyncio
+import json
+import logging
+
+from aiohttp import web
+from aiopg.sa import Engine
+from sqlalchemy.sql import select
+
+from servicelib.application_keys import APP_DB_ENGINE_KEY
+
+from .computation_config import APP_COMP_TASKS_LISTENING_KEY
+from .projects import projects_api
+from .projects.projects_models import projects, user_to_projects
+
+log = logging.getLogger(__name__)
+
+DB_PROCEDURE_NAME: str = "notify_comp_tasks_changed"
+DB_TRIGGER_NAME: str = f"{DB_PROCEDURE_NAME}_event"
+DB_CHANNEL_NAME: str = "comp_tasks_output_events"
+async def register_trigger_function(app: web.Application):
+    db_engine: Engine = app[APP_DB_ENGINE_KEY]
+    # NOTE: an example was found in https://citizen428.net/blog/asynchronous-notifications-in-postgres/
```

Comment:
Check how to do this directly iwht aiopg https://aiopg.readthedocs.io/en/stable/examples.html#usage-of-listen-notify-commands

Reply:
Yes that is what is done (see line 76) with the listen function.
As is shown in [this link](https://citizen428.net/blog/asynchronous-notifications-in-postgres/) it is not sufficient to just "listen" to the events. The database needs some programming as well, which is what **register_trigger_function** does. A fct needs to be registered within the database + a TRIGGER needs to be defined [see here](https://www.postgresql.org/docs/9.1/sql-createtrigger.html).
This should be created somehow through alembic I guess?